cluster/interface.go

```diff
@@ -0,0 +1,19 @@
+package cluster
+
+const (
+	// EtcdLeaderElector is the id used when constructing an
+	// etcdLeaderElector instance through the factory.
+	EtcdLeaderElector = "etcdLeaderElector"
+)
+
+// LeaderElector is a general interface implementing basic leader elections
+// in a clustered environment.
+type LeaderElector interface {
```

Comment:
One other thing that would be nice to have would be the ability to query if a node is the leader or not. I think with the latest set of RPC changes by Johan, the main RPC server should be active at all times. If we have such an endpoint, it may also be able to eventually be used as a health check for laod balancer to query which of the active lnd instances is actually the leader.

Reply:
Added a new commit to include a "waiting to start" state in the state RPC which can be used to query if an instance has become the leader.